pyro/distributions/sine_skewed.py

```diff
@@ -0,0 +1,92 @@
+    :class:`~pyro.infer.HMC` or :class:`~pyro.infer.NUTS`. For example, the following will produce a uniform prior
+    over skewness for the 1-torus,::
+
+        def model(...):
+            ...
+            skewness_phi = pyro.sample(f'skewness_phi', Uniform(skewness.abs().sum(), 1 - tots))
+            psi_bound = 1 - skewness_phi.abs()
+            skewness_psi = pyro.sample(f'skewness_psi', Uniform(-psi_bound, psi_bound)
+            skewness = torch.stack((skewness_phi, skewness_psi), dim=0)
+            ...
+
+    .. note:: An event in the base distribution must be on a d-torus, so the event_shape must be (d, 2) or (2,).
+
+    .. note:: For the skewness parameter, it must hold that the sum of the absolute value of its weights for an event
+        must be less than or equal to one. See eq. 2.1 in [1].
+
+    ** References: **
+      1. Sine-skewed toroidal distributions and their application in protein bioinformatics
+         Ameijeiras-Alonso, J., Ley, C. (2019)
+
+    :param base_density: base density on a d-dimensional torus.
+    :param skewness: skewness of the distribution.
+    """
+    arg_constraints = {'skewness': constraints.independent(constraints.interval(-1., 1.), 1)}
+
+    support = constraints.independent(constraints.real, 1)
+
+    def __init__(self, base_density: TorchDistribution, skewness, validate_args=None):
+        assert base_density.event_shape[-1] == 2 and len(base_density.event_shape) <= 2
+        assert base_density.shape()[len(base_density.shape()) - len(skewness.shape):] == skewness.shape
+        assert (skewness.abs().sum(-1 if len(skewness.shape) == 1 else (-2, -1)) <= 1).all()
+
+        self.base_density = base_density
+        self.skewness = skewness.broadcast_to(base_density.shape())
+        super().__init__(base_density.batch_shape, base_density.event_shape, validate_args=validate_args)
+
+        if self._validate_args and base_density.mean.device != skewness.device:
+            raise ValueError(f"base_density: {base_density.__class__.__name__} and SineSkewed "
+                             f"must be on same device.")
+
+    def __repr__(self):
+        args_string = ', '.join(['{}: {}'.format(p, getattr(self, p)
+                                if getattr(self, p).numel() == 1
+                                else getattr(self, p).size()) for p in self.arg_constraints.keys()])
+        return self.__class__.__name__ + '(' + f'base_density: {str(self.base_density)}, ' + args_string + ')'
+
+    def sample(self, sample_shape=torch.Size()):
+        bd = self.base_density
+        ys = bd.sample(sample_shape)
+        u = Uniform(0., torch.ones(torch.Size([]), device=self.skewness.device)).sample(sample_shape + self.batch_shape)
+
+        mask = u < 1. + (self.skewness * torch.sin((ys - bd.mean) % (2 * pi))).view(*(sample_shape + self.batch_shape),
+                                                                                    -1).sum(-1)
```

Comment:
I think you can use `pyro.distributions.util.sum_rightmost`
```
mask = u < 1. + sum_rightmost(self.skewness * torch.sin((ys - bd.mean) % (2 * pi)), self.event_dim)
```

Reply:
Cool, thanks. I'll try it out.